Libraries/Utilities/deprecatedCallback.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule deprecatedCallback
+ * @flow
+ */
+
+'use strict';
+
+// @deprecated, will be removed on April 1 2016
+// helper for deprecated callbacks pattern
+module.exports = function(promise: Promise, callbacks: Array<Function>, type: string, warning: string): Promise {
+  if (callbacks.length > 0) {
+    let success, error;
+
+    console.warn(warning);
+
+    switch (type) {
+    case 'success-first': // handles func(success, error), func(success)
+      [ success, error ] = callbacks;
+      return promise.then(
+        res => success(res),
+        err => {
+          if (error) {
+            error(err);
+          } else {
+            throw err;
```

Comment:
Is throw the right thing to do here?  Or should we just silently swallow it or maybe console.warn that the user had an error they are ignoring?  Throwing will potentially change behavior which we don't want to do.